### PR TITLE
Fix when using empty settings

### PR DESCRIPTION
**Pull request by jorisvaesen**
Fix for #371

### CODE REVIEW

File: src/Model/Behavior/UploadBehavior.php [hunk 1]
```diff
@@ -74,6 +75,8 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
                 continue;
             }
 
+            $settings = is_array($settings) ? $settings : [];
```
Comment (josegonzalez) [inline src/Model/Behavior/UploadBehavior.php]: Why not properly setup the settings in the initialize?

Comment (jorisvaesen) [inline src/Model/Behavior/UploadBehavior.php]: Was thinking the same, just like Hash::normalize too much :)

Comment (josegonzalez) [inline src/Model/Behavior/UploadBehavior.php]: So after the `Hash::normalize()` call, detect an invalid config and fix it :)

Comment (jorisvaesen) [inline src/Model/Behavior/UploadBehavior.php]: Nah, we'd better loop the config ourselfs, otherwise it's doing the job twice.

**Comment by kevinmasson** [comment]
The error is till happening for me. I'm using 3.6.2 version.

**Error :** 
`InvalidArgumentException: Invalid data type, must be an array or \ArrayAccess instance. `

**This works :** 
```
$this->addBehavior('Josegonzalez/Upload.Upload', [
	   'thumbnail' => [], 
	]);
```
**This doesn't work :** 
```
$this->addBehavior('Josegonzalez/Upload.Upload', [
	   'thumbnail'
	]);
```
**Stacktrace :**
Stacktrace

**Comment by josegonzalez** [comment]
@kevinmasson mind filing an issue?

**Pull request merged**